### PR TITLE
[DCP] Make fsspec public

### PR DESCRIPTION
**Pull request by LucasLLC**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121329

Makes  fsspec public

We're going to merge this in https://github.com/pytorch/pytorch/pull/121508 instead due to more CI issues. 

Differential Revision: [D54596394](https://our.internmc.facebook.com/intern/diff/D54596394/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

### CODE REVIEW

File: torch/distributed/checkpoint/_fsspec_filesystem.py [hunk 1]
```diff
@@ -1,122 +1,15 @@
...
-    def __init__(self, path: Union[str, os.PathLike]) -> None:
-        super().__init__(path)
-        self.fs = FileSystem()
-        self.path = self.fs.init_path(path)
-
-    @classmethod
-    def validate_checkpoint_id(cls, checkpoint_id: Union[str, os.PathLike]) -> bool:
-        return FileSystem.check(checkpoint_id)
+log = logging.getLogger(__name__)
+log.warning(
```
Comment (wz337) [inline torch/distributed/checkpoint/_fsspec_filesystem.py:12]: Noice!

**Comment by wz337** [comment]
Discussed offline for public API naming!

**Comment by LucasLLC** [comment]
> Discussed offline for public API naming!

Renamed `fsspec_filesystem.py` -> `fsspec.py` as discussed -- thanks for catching this!

**Review by wz337 — APPROVED** [review]
Thanks for making the changes! LGTM!

**Review by fegin — APPROVED** [review]
Overall, this looks good to me. But I'm not sure if it is necessary to expose the default storage (same question for planner).

**Comment by LucasLLC** [comment]
> Overall, this looks good to me. But I'm not sure if it is necessary to expose the default storage (same question for planner).

I'm not sure I'm following, could you point out the line to me?

**Pull request closed without merging**